src/View/Shortcodes/EmbedShortcodeProvider.php

```diff
@@ -89,7 +89,24 @@ public static function handle_shortcode($arguments, $content, $parser, $shortcod
         }
 
         // Process embed
-        $embed = $embed->getEmbed();
+        try {
+            $embed = $embed->getEmbed();
+        }
+        catch(\Embed\Exceptions\InvalidUrlException $e) {
+            $attr = array(
+                'class' => 'ss-media-exception embed'
+            );
+            if (Director::isDev()) {
+                $result = HTML::createTag('div', $attr, $e->getMessage());
+                return $result;
+            }
+            $result = HTML::createTag(
+                'div',
+                $attr,
+                HTML::createTag('p', array(), 'There was a problem loading the media.')
+            );
```

Comment:
As much as I like early returns, this one seems a little out of place? Why not just reuse the same generated tag for both dev and non-dev, and just swap out the message?

Reply:
Looks like this was implemented